src/scripts/main.js

```diff
@@ -1,3 +1,37 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+
+document.addEventListener('click', (e) => {
+  const target = e.target.closest('thead th');
+
+  if (!target) {
+    return;
+  }
+  sort(target);
+});
+
+function sort(elem) {
+  // we find the column index number
+  const thead = [...table.rows[0].cells];
+  const currentPos = thead.indexOf(elem);
+
+  const tableBody = [...table.rows].slice(1, -1).sort((a, b) => {
+    let firstValue = a.cells[currentPos].innerHTML;
+    let secondValue = b.cells[currentPos].innerHTML;
+
+    // we check whether these are numbers
+    if (parseFloat(firstValue) || firstValue.includes('$')) {
```

Comment:
The condition parseFloat(firstValue) is incorrect for checking if a value is a number. It will return NaN for non-numeric strings, which is falsy, but it will also return 0 for '0', which is falsy as well. Consider using !isNaN(parseFloat(firstValue)) instead.